controllers/iamrole_reconcile_manager.go

```diff
@@ -0,0 +1,68 @@
+package controllers
+
+import (
+	"context"
+	"time"
+
+	api "github.com/keikoproj/iam-manager/api/v1alpha1"
+	"github.com/keikoproj/iam-manager/internal/config"
+	"github.com/keikoproj/iam-manager/pkg/logging"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+/**
+ * This will start a go routine that will run every "ControllerDesiredFrequency" seconds.
+ */
+func (r *IamroleReconciler) StartControllerReconcileCronJob(ctx context.Context) error {
+	log := logging.Logger(context.Background(), "controllers", "iamrole_controller", "StartControllerReconcileCronJob")
+
+	// Kick off a go routine to reconcile all the IAM roles
+	var err error
+	var iamRoles []*api.Iamrole
+
+	log.Info("Starting the cronjob")
+
+	tick := time.NewTicker(time.Duration(config.Props.ControllerDesiredFrequency()) * time.Second)
+
+	for _ = range tick.C {
```

Comment:
Test that does what you expect, you may want to switch to a go routine now, as long as it's blocking and can use same context and wait for Done.

Ex.
```go

	go func() {
		for {
			select {
			case <-ctx.Done:
				return
			case tm := <-tick.C:
				...
			}
		}
	}()
```

Reply:
nvm, this as I see you tested it, let's just make sure you handle `ctx.Done`.

Reply:
> nvm, this as I see you tested it, let's just make sure you handle `ctx.Done`.

Updated logic to handle ctx.Done() channel event

Reply:
Verified if reconcile time elapsed longer than frequency, ticker channel will not accumulate events.